src/interfaces/chain.h

```diff
@@ -217,6 +236,36 @@ class Chain
 
     //! Send wallet load notification to the GUI.
     virtual void loadWallet(std::unique_ptr<Wallet> wallet) = 0;
+
+    //! Send progress indicator.
+    virtual void showProgress(const std::string& title, int progress, bool resume_possible) = 0;
+
+    //! Chain notifications.
+    class Notifications
+    {
+    public:
+        virtual ~Notifications() {}
+        virtual void TransactionAddedToMempool(const CTransactionRef& tx) {}
+        virtual void TransactionRemovedFromMempool(const CTransactionRef& ptx) {}
+        virtual void BlockConnected(const CBlock& block,
```

Comment:
Can you simplify this interface to `(cont CBlock& block, const std::vector<CTransactionRef>& tx_conflicted)`? The client can get the block hash using `CBlock->GetHash()`.